elastic/kibana
# [Fleet] replaced upgraded_at null by upgrade_status

## Summary

Fix https://github.com/elastic/kibana/issues/139704

Using a new field `upgrade_status` instead of setting `upgraded_at` to `null` in Fleet agents, to workaround a lucene bug that sometimes gives an error to `/agent_status` query.

## To test:
1. Enroll fleet server, and many agents with horde
2. Bulk select agents and upgrade immediately
3. Wait for upgrade to complete, notice that agents are gradually showing up as `Healthy` and the Upgrade callout decreases in count.
4. Expect that the lucene error does not come anymore on the `agent_status` API call

See in the recording, tested with 25 agents (FS is unhealthy, this is a known issue with running it locally). After starting upgrade with 1 hour window, 1 agent completes immediately (goes back to Healthy), the rest remain in Updating. After abort, all agents go back to Healthy.

https://user-images.githubusercontent.com/90178898/188823348-371296c6-1450-4925-849c-651b71746456.mov

<img width="1781" alt="image" src="https://user-images.githubusercontent.com/90178898/188825166-f50618e8-8445-403e-a3ca-fb4149c5485e.png">


### Checklist
